generated_code/NodalBoundaryConditions.py

```diff
@@ -53,7 +54,10 @@ def addKernels(generator, aderdg, include_tensors, matricesDir, dynamicRuptureMe
     # To be used as Tinv in flux solver - this way we can save two rotations for
     # Dirichlet boundary, as ghost cell dofs are already rotated
     identity_rotation = np.double(aderdg.transformation_spp())
-    identity_rotation[0:9, 0:9] = np.eye(9)
+    if isinstance(aderdg, acoustic.AcousticADERDG):
```

Comment:
That works well.

However, you can even go a bit further and set `identity_rotation[0:quantities, 0:quantities] = np.eye(quantities)`, where `quantities = aderdg.numberOfQuantities()`.

(we do have some material with 13 quantities in—that way, we'd cover it here as well already)